utils/core/src/serde/byte_writer.rs

```diff
@@ -0,0 +1,80 @@
+use super::{Serializable, Vec};
+
+// BYTE WRITER TRAIT
+// ================================================================================================
+
+/// Defines how primitive values are to be written into `Self`.
+pub trait ByteWriter: Sized {
+    // REQUIRED METHODS
+    // --------------------------------------------------------------------------------------------
+
+    /// Writes a single byte into `self`.
+    ///
+    /// # Panics
+    /// Panics if the byte could not be written into `self`.
+    fn write_u8(&mut self, value: u8);
+
+    /// Writes a sequence of bytes into `self`.
+    ///
+    /// # Panics
+    /// Panics if the sequence of bytes could not be written into `self`.
+    fn write_bytes(&mut self, values: &[u8]);
```

Comment:
This is debatable, but we might want it to be fallible instead of potentially throwing a panic.

If we plan to have only `Vec` as implementor, then we don't really need to worry about that as the only possible failure is when we run out of memory. However, we might want in the future to send serializable data to I/O (either network, disk, some big table, etc). If the trait is infallible, we won't allow such options.

Reply:
Agreed - though this will introduce more changes than I'd like to handle in this PR. So, leaving it for the future.